pkg/plugins/runtime/k8s/webhooks/resourceadmissionchecker.go

```diff
@@ -41,6 +42,10 @@ func (c *ResourceAdmissionChecker) IsOperationAllowed(userInfo authenticationv1.
 		return c.resourceIsNotAllowedResponse()
 	}
 
+	if errResponse := c.validateLabels(r); errResponse != nil {
```

Comment:
The method `IsOperationAllowed` calls `isResourceTypeAllowed` to check if "TypeMeta" (what we call Descriptor) is allowed in the environment, then it calls `isResourceAllowed` to check that ResourceMeta+ResourceSpec are allowed in the environment. Your call to `validateLabels` should be moved inside `isResourceAllowed` as it checks ResourceMeta

Reply:
moved